packages/core/src/render3/query.ts

```diff
@@ -133,6 +137,20 @@ function readFromNodeInjector(nodeInjector: LNodeInjector, node: LNode, read: Qu
   }
 }
 
+function getDirectiveMatchingSelector(node: LNode, selector: string): any|null {
+  const ngData = node.view.data;
+  const flags = node.flags;
+  for (let i = flags >> LNodeFlags.INDX_SHIFT,
+           ii = i + ((flags & LNodeFlags.SIZE_MASK) >> LNodeFlags.SIZE_SHIFT);
+       i < ii; i++) {
+    const instance = ngData[i];
+    if (instance[NG_LOCALNAME_SYMBOL] === selector) {
```

Comment:
this is a megamorphic read, hence slow. Need to solve it differently. We probably need to store it on the staticData